File: 0875-koko-eating-bananas/0875-koko-eating-bananas.js
/**
 * @param {number[]} piles
 * @param {number} h
 * @return {number}
 */
var minEatingSpeed = function(piles, h) {
    let low=1;
    let high = Math.max(...piles);
    let ans;
    while(low<=high){
        let mid= Math.floor((low+high)/2);
        let hours=0;
        for(let i=0;i<piles.length;i++){
            hours += Math.ceil(piles[i]/mid)
        }
        if (hours<=h){
            ans= mid;
            high=mid-1
        }
        else{
            low = mid+1;
        }
    }
    return ans
    
};

File: 0875-koko-eating-bananas/0875-koko-eating-bananas.cpp
class Solution {
public:
    long long int calculate(vector<int>& piles, long long int h,
                            long long int mid) {
        long long int total = 0;
        for (int i = 0; i < piles.size(); i++) {
            long long int rem = piles[i] % mid;
            if (rem != 0) {
                total++;
            }
            total = total + (piles[i] / mid);
        }
        if (total <= h)
            return true;
        return false;
    }
    int minEatingSpeed(vector<int>& piles, int h) {
        long long int low = 0;
        long long int high = 0;

        for (int i = 0; i < piles.size(); i++) {
            
            high = high + piles[i];
        }
        long long int ans = -1;
        while (low <= high) {
            long long int mid = (low + high) / 2;
            cout <<low<< mid << endl;
            if (calculate(piles, h, mid)) {
                ans = mid;
                high = mid - 1;
            } else {
                low = mid + 1;
            }
        }
        return ans;
    }
};